scripts/valhalla_build_elevation

```diff
@@ -237,39 +244,69 @@ def download(tile: Tile, output_dir, compression: TileCompression):
     url = f"https://elevation-tiles-prod.s3.us-east-1.amazonaws.com/skadi/{tile.dir}/{tile.name}.gz"
 
     LOGGER.info(f"Downloading tile {tile.name}")
-    try:
-        with request.urlopen(url) as res, open(filepath, "wb") as f:
-            if compression is TileCompression.GZIP:
-                f.write(res.read())
-            else:
-                with gzip.GzipFile(fileobj=res, mode="rb") as gz:
-                    uncompressed = gz.read()
-                    if compression is TileCompression.UNCOMPRESSED:
-                        f.write(uncompressed)
-                    elif compression is TileCompression.LZ4:
-                        # Compression level 6 was chosen after some benchmarking as the approx efficient frontier
-                        # between compression time and space savings (decompression time is roughly constant regardless
-                        # of level). The end result is larger than the maximally gzipped tiles from AWS, but only
-                        # by around 12%.
-                        import lz4.frame
-
-                        with lz4.frame.LZ4FrameCompressor(
-                            block_size=lz4.frame.BLOCKSIZE_MAX4MB, compression_level=6
-                        ) as compressor:
-                            # Optimization: we know the exact size of every uncompressed hgt file
-                            f.write(compressor.begin(25934402))
-                            f.write(compressor.compress(uncompressed))
-                            f.write(compressor.flush())
-
-            LOGGER.debug(f"Successfully downloaded tile {tile.name}")
-
-        return True
-    except URLError as e:
-        LOGGER.critical(f"Download failed of elevation tile {tile.dir}/{tile.name}: {e.reason}")
+
+    download_status = DownloadStatus.FAILED
```

Comment:
most of the diff is just wrapping the previous logic into a for loop, where we request 15 times maximum with increased sleeps in between (exponential back off), up to 4.5 hours until we give up. could give much earlier though :D